js/source/scripts.js

```diff
@@ -10,27 +10,128 @@
 // wrapping it with an "anonymous closure". See:
 // - https://drupal.org/node/1446420
 // - http://www.adequatelygood.com/2010/3/JavaScript-Module-Pattern-In-Depth
-(function (Drupal, $, window) {
+ 
+  // Debounce https://davidwalsh.name/javascript-debounce-function
+ (function ($) {
+
+  function debounce(func, wait, immediate) {
```

Comment:
Drupal core has this function. I am going to remove it from here as it duplicates a library that is already available.